src/classes/PREN_ProgramPlan_TDTM.cls

```diff
@@ -0,0 +1,103 @@
+
+    /*******************************************************************************************************
+    * @description Handles Program Plan management for Program Enrollment.
+    * @param listNew the list of Program Enrollments from trigger new.
+    * @param listOld the list of Program Enrollments from trigger old.
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @param objResult the describe for Program Enrollment.
+    * @return dmlWrapper.
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
+            TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+                
+        //Before Insert
+        if (newlist != null
+            	&& newlist.size() > 0
+            	&& triggerAction == TDTM_Runnable.Action.BeforeInsert 
+            	&& !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PREN_ProgramPlan_TDTM_Before_Insert)) {
+            if(newlist != null && newlist.size() > 0) {
+                populatePrimaryProgramPlan(newlist);
+            }
+        }
+                
+		if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.PREN_ProgramPlan_TDTM_Before_Insert, false);
+        }
+        return dmlWrapper;
+    }
+
+    /*******************************************************************************************************
+    * @description Populate primary program plan when a program enrollment is created
+    * @param List<SObject> newList the Contact
+    * @return void
+    */
+    private void populatePrimaryProgramPlan(List<SObject> newlist) {
+        Set<Id> setAcademicProgramIds = new Set<Id>();
+        Map<Id, Id> mapAcademicProgramIdToPrimaryPPId = new Map<Id, Id>();
+        for (SObject so : newlist) {
+            Program_Enrollment__c newPe = (Program_Enrollment__c)so;
+            //Build a set of academic program ids
+            setAcademicProgramIds.add(newPe.Account__c);
+        }
+
+        //Query academic program and build a map of
+        //Academic Program id and Primary Program Plan id
+        for (Program_Plan__c pp : [SELECT Id, Account__c
+                                    FROM Program_Plan__c
+                                    WHERE Account__c in :setAcademicProgramIds
+                                        AND Is_Primary__c = TRUE]) {
+            if (pp.Account__c != null) {
+                mapAcademicProgramIdToPrimaryPPId.put(pp.Account__c, pp.Id);
+            }
+        }
+
+        //Loop through program enrollment again and populate primary program plan
+        for (SObject so : newlist) {
+            Program_Enrollment__c newPe = (Program_Enrollment__c) so;
```

Comment:
did you intend to have the space here? I like the way you formatted line 77. Can we do the same here?